cluster/swarm/cluster.go

```diff
@@ -158,10 +163,39 @@ func (c *Cluster) hasEngineByAddr(addr string) bool {
 	return c.getEngineByAddr(addr) != nil
 }
 
-func (c *Cluster) addEngine(addr string) bool {
+func (c *Cluster) addEngine(addr string) {
+	res, engine := c.registEngine(addr)
+	if !res && engine == nil {
+		//		ticker := time.Tick(reAddEnginePeriod)
```

Comment:
@vieux I think the comments way maybe better, but I imitate the code in "discovery/file/file.go" from 83 line to 100 line, to maintain the coding style.